perfkitbenchmarker/providers/ibmcloud/util.py

```diff
@@ -0,0 +1,212 @@
+      route_entry = subnet_cidr_block.split('/24')
+      interface = None
+      for line in data.splitlines():
+        items = line.split()
+        if len(items) > 6 and items[0] == route_entry[0]:
+          interface = items[7]
+          route_cmds.append('ip route add ' + target_cidr_block + ' via ' + \
+                            subnet_gateway + ' dev ' + interface)
+  return route_cmds
+
+
+def GetMinMax(data):
+  """Finds min and max values in the given list of values"""
+  if not data:
+    return 0.0, 0.0
+  min_value = float(data[0])
+  max_value = float(data[0])
+  for item in data:
+    value = float(item)
+    if value > max_value:
+      max_value = value
+    if value < min_value:
+      min_value = value
+  return min_value, max_value
+
+
+def GetBaseOs(osdata):
+  """Finds the base os name to use
+
+  Args:
+    osdata: json representation of os information of an image.
+
+  Returns:
+    Short name of the os name
+  """
+  if 'name' in osdata and osdata['name'] is not None:
+    osname = osdata['name'].lower()
+    if 'window' in osname:
+      return WINDOWS
+    elif 'red' in osname or 'rhel' in osname:
+      return REDHAT
+    elif 'centos' in osname:
+      return CENTOS
+    elif 'debian' in osname:
+      return DEBIAN
+    elif 'ubuntu' in osname:
+      return UBUNTU
+  return UNKNOWN
+
+
+def GetGen(account):
+  """Creates a ibmcloud access object """
+  gen = ibmcloud.IbmCloud(account=account[0], apikey=account[1], verbose=False, \
+                     version='v1', silent=True, force=True)
+  if not gen.token():
+    gen.set_token()  # one more try
+  return gen
+
+
+def GetImageId(account, imgname):
+  """Returns image id matching the image name """
+  data_mgr = ibmcloud.ImageManager(GetGen(account))
+  resp = data_mgr.List()['images']
+  if resp is not None:
+    for image in resp:
+      if image['name'] == imgname:
+        return image['id']
+  return None
+
+
+def GetImageIdInfo(account, imageid):
+  """Returns OS information matching the image id """
+  data_mgr = ibmcloud.ImageManager(GetGen(account))
+  return GetOsInfo(data_mgr.Show(imageid))
+
+
+def GetOsInfo(image):
+  """Returns os information in json format
+
+  Args:
+    image: json representation of os information of an image.
+
+  Returns:
+    OS details of the image in json format
+  """
+  data = {}
+  custom = False
+  if image and 'operating_system' in image:
+    data = image['operating_system']
+    if 'href' in data:
+      del data['href']  # delete this, does not seem necessary
+    if 'custom' in image['name']:  # this name is not the name in operating_system
+      custom = True
+  else:
+    # if lookup failed, try read from env if any is set
+    data['version'] = os.environ.get('RIAS_IMAGE_OS_VERSION')
+    data['vendor'] = os.environ.get('RIAS_IMAGE_OS_VENDOR')
+    data['name'] = os.environ.get('IMGNAME')
+    if data['name'] is not None and 'custom' in data['name']:
```

Comment:
can just say `if 'custom' in data['name'] or ''`  that is assuming you want `data['name'] = None` to be a value.

Reply:
data['name'] can be None,
if 'custom' in data['name'] or ''
this fails so I am leaving as is.